test/session_tests/DocumentQueryTest.ts

```diff
@@ -8,220 +8,493 @@ import {RequestExecutor} from "../../src/Http/Request/RequestExecutor";
+
+      expect(results).to.have.lengthOf(10);
+
+        results.map(function(obj) {
+
+          expect(obj).to.include({'name': 'withNesting'});
+
+        });
+
     });
 
-    it('should fail query with non-existing index', async () => expect(
-        store.openSession().query({
-          indexName: 's'
-        }).where({'Tag': 'Products'}).get()
-      ).to.be.rejected
+    it('should query with whereLessThanOrEqual', async () => {
+      const results: Universal[] = await  store.openSession()
+        .query<Universal>({
+          indexName: 'Universals',
+          WaitForNonStaleResults: true
+        })
+        .whereLessThanOrEqual<number>('order',6, 'name', 'withNesting').get();
+
+      expect(results).to.have.lengthOf(10);
+
+      results.map(function(obj) {
+
+        expect(obj).to.include({'name': 'withNesting'});
+
+      });
+
+    });
+
+    it('should query with nested objects', async () => {
+      const results: Universal[] = await store.openSession().query<Universal>({
+        indexName: 'Universals',
+        nestedObjectTypes: {product: Product}
+      }).whereEquals<string>('name', 'withNesting').get();
+
+      expect(results).to.have.lengthOf(10);
+
+      results.map(function (obj) {
+        expect(obj.name).to.equal('withNesting');
+        expect(obj.product.name).to.equal('testing_order');
+      });
+
+    });
+
+    it('should make query with fetch terms', async () => {
+      const results: Universal[] = await store.openSession().query<Universal>({
+        WaitForNonStaleResults: true,
+        indexName: 'Universals'
+      }).selectFields('id').get();
+
+      expect(results).to.have.lengthOf(10);
+      expect(_.every(results, (result: Product) => result.hasOwnProperty('id'))).to.be.true;
+
+    });
+
+    it('should query with whereArray', async () => expect(
+      store.openSession().query<Universal>({
+        indexName: 'Universals'
+      }).where({
+        name: [4, 6, 90]
+      }).get()
+      ).to.be.fulfilled
     );
 
-    it('should query with where', async() => expect(
-        store.openSession().query().where({
-          name: 'test101', 
-          uid: [4, 6, 90]
-        }).get()
+    it('should query with where', async () => expect(
+      store.openSession().query<Universal>({
+        indexName: 'Universals'
+      }).where({name: 4}).get()
       ).to.be.fulfilled
     );
 
-    it('should query with index', async() => {
-      const results: Product[] = await store.openSession().query<Product>({
-        documentType: Product,
-        indexName: 'Testing_Sort',
-        waitForNonStaleResults: true
-      }).where({
-        uid: [4, 6, 90]
-      }).get();
+    it('should fail query with non-existing index', async () => {
+      const results = await store.openSession().query<Universal>({
+        indexName: 'none'
+      }).whereIn('Tag', 'Products').get();
+
+      expect(results).to.have.lengthOf(0);
+
+    });
+
+    it('should query by @all_docs index', async () => {
```

Comment:
Either test description or its body is wrong. I can't see reference to `@all_docs` anywhere here.